### ISSUE TITLE
🐛 Bug Report: The Appwrite SDK does not connect appropriately using Docker

### ISSUE THREAD

**Issue by kelton-glitch** [appwrite/appwrite#7130]
### 👟 Reproduction steps

So I used Appwrite a few months ago and the connection process was to set up the port through which the SDK could connect and display through a web browser. I went back to an old project and decided to set it up with appwrite and I followed the [installation process using Docker](https://appwrite.io/docs/advanced/self-hosting) which worked well.

### 👍 Expected behavior

I expected the port to connect automatically based on the 80:80 default port configuration

### 👎 Actual Behavior

It actually returns an error message saying Internal server error message
```
"message": "Error: Server Error",
  "code": 500,
  "version": "1.4.9"
```

I tried changing the default ports severally but maintaining the TCP protocol, but I seem not to have found a solution yet

### 🎲 Appwrite version

Version 1.4.x

### 💻 Operating system

Windows

### 🧱 Your Environment

I used Flutter for the development of an application and I am using Appwrite as the database

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by kelton-glitch** [comment]
[Error] Type: Utopia\Exception
2023-11-15 09:56:44 [Error] Message: Error handler had an error: Failed to create connection: php_network_getaddresses: getaddrinfo failed: Try again
2023-11-15 09:56:44 [Error] File: /usr/src/code/vendor/utopia-php/framework/src/App.php
2023-11-15 09:56:44 [Error] Line: 551
2023-11-15 09:56:50 Database not ready. Retrying connection (6)...
2023-11-15 09:57:08 Database not ready. Retrying connection (7)...
2023-11-15 09:57:19 [Error] Type: Utopia\Exception
2023-11-15 09:57:19 [Error] Message: Error handler had an error: Failed to create connection: php_network_getaddresses: getaddrinfo failed: Try again
2023-11-15 09:57:19 [Error] File: /usr/src/code/vendor/utopia-php/framework/src/App.php
2023-11-15 09:57:19 [Error] Line: 551
2023-11-15 09:57:27 Database not ready. Retrying connection (8)...
2023-11-15 09:57:45 Database not ready. Retrying connection (9)...
2023-11-15 09:58:03 Database not ready. Retrying connection (10)...
2023-11-15 09:58:03 
2023-11-15 09:58:03 Fatal error: Uncaught Exception: Failed to connect to database: Failed to create connection: php_network_getaddresses: getaddrinfo failed: Try again in /usr/src/code/app/http.php:82
2023-11-15 09:58:03 Stack trace:
2023-11-15 09:58:03 #0 {main}
2023-11-15 09:58:03   thrown in /usr/src/code/app/http.php on line 82
2023-11-15 09:58:03 [2023-11-15 08:58:03 *8.0]  ERROR   php_swoole_server_rshutdown() (ERRNO 503): Fatal error: Uncaught Exception: Failed to connect to database: Failed to create connection: php_network_getaddresses: getaddrinfo failed: Try again in /usr/src/code/app/http.php:82
2023-11-15 09:58:03 Stack trace:
2023-11-15 09:58:03 #0 {main}
2023-11-15 09:58:03   thrown in /usr/src/code/app/http.php on line 82
2023-11-15 09:58:20 Database not ready. Retrying connection **(1)...**


This is an excerpt of what shows in the Docker logs when I try to access appwrite

**Comment by stnguyen90** [comment]
@kelton-glitch, some additional questions:

1. What's the output of `docker ps`?
2. Did you change DB related env vars after installation?
3. What's the output of `appwrite_appwrite-mariadb`. Does the CreatedAt timestamp match when you installed Appwrite?

**Comment by CoderoidAcademy** [comment]
this is docker ps

root@ubuntu-g2-small2-bamdad-1:/home/ubuntu# docker ps
CONTAINER ID   IMAGE                         COMMAND                  CREATED       STATUS          PORTS                                                                      NAMES
59a82dc3737a   traefik:2.9                   "/entrypoint.sh --pr…"   2 hours ago   Up 2 hours      0.0.0.0:80->80/tcp, :::80->80/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp   appwrite-traefik
bec4e5cf848a   appwrite/appwrite:1.4.9       "usage"                  2 hours ago   Up 13 seconds   80/tcp                                                                     appwrite-usage
4061bb5ec545   appwrite/appwrite:1.4.9       "worker-databases"       2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-databases
806f198e859d   appwrite/appwrite:1.4.9       "schedule"               2 hours ago   Up 13 seconds   80/tcp                                                                     appwrite-schedule
1f4f98cf2a44   appwrite/appwrite:1.4.9       "worker-audits"          2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-audits
47070d325081   appwrite/appwrite:1.4.9       "realtime"               2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-realtime
1c8ff04ddb92   appwrite/appwrite:1.4.9       "worker-certificates"    2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-certificates
a0c8d2c735f5   appwrite/appwrite:1.4.9       "worker-webhooks"        2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-webhooks
a824cb88e615   appwrite/appwrite:1.4.9       "maintenance"            2 hours ago   Up 13 seconds   80/tcp                                                                     appwrite-maintenance
265c83b074e2   appwrite/appwrite:1.4.9       "worker-functions"       2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-functions
6be189b6182b   appwrite/appwrite:1.4.9       "docker-php-entrypoi…"   2 hours ago   Up 2 hours      80/tcp                                                                     appwrite
d7b3f77d6837   appwrite/appwrite:1.4.9       "worker-mails"           2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-mails
fcf807c28ce9   appwrite/appwrite:1.4.9       "worker-messaging"       2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-messaging
aa817339f29e   appwrite/appwrite:1.4.9       "worker-deletes"         2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-deletes
605e260ffacc   appwrite/appwrite:1.4.9       "worker-builds"          2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-builds
77404e53f8c2   appwrite/appwrite:1.4.9       "worker-migrations"      2 hours ago   Up 2 hours      80/tcp                                                                     appwrite-worker-migrations
8fd90a92c7d7   appwrite/telegraf:1.4.0       "/entrypoint.sh tele…"   2 hours ago   Up 2 hours      8092/udp, 8125/udp, 8094/tcp                                               appwrite-telegraf
6de1b020c367   appwrite/influxdb:1.5.0       "/entrypoint.sh infl…"   2 hours ago   Up 2 hours      8086/tcp                                                                   appwrite-influxdb
d590dbceeedb   appwrite/assistant:0.2.2      "docker-entrypoint.s…"   2 hours ago   Up 2 hours      3003/tcp                                                                   appwrite-assistant
3594f6b1fb14   openruntimes/executor:0.4.5   "docker-php-entrypoi…"   2 hours ago   Up 2 hours      80/tcp                                                                     openruntimes-executor
e5b92992a0c3   mariadb:10.7                  "docker-entrypoint.s…"   2 hours ago   Up 2 hours      3306/tcp                                                                   appwrite-mariadb
e9548585fc0f   redis:7.0.4-alpine            "docker-entrypoint.s…"   2 hours ago   Up 2 hours      6379/tcp                                                                   appwrite-redis
root@ubuntu-g2-small2-bamdad-1:/home/ubuntu#

**Comment by CoderoidAcademy** [comment]
about above comment:

i have this problem, too.
i install docker and after install appwrite. all level done successful. but in browser, i get this output (in attach file)
![Capture2](https://github.com/appwrite/appwrite/assets/33560157/39b6b84b-2f40-4363-acc7-9609111c7250)

**Comment by CoderoidAcademy** [comment]
**docker --version and docker --status:**

root@ubuntu-g2-small2-bamdad-1:/home/ubuntu# docker --version
Docker version 24.0.7, build afdd53b
root@ubuntu-g2-small2-bamdad-1:/home/ubuntu# docker --status
unknown flag: --status
See 'docker --help'.

Usage:  docker [OPTIONS] COMMAND

A self-sufficient runtime for containers

Common Commands:
  run         Create and run a new container from an image
  exec        Execute a command in a running container
  ps          List containers
  build       Build an image from a Dockerfile
  pull        Download an image from a registry
  push        Upload an image to a registry
  images      List images
  login       Log in to a registry
  logout      Log out from a registry
  search      Search Docker Hub for images
  version     Show the Docker version information
  info        Display system-wide information

Management Commands:
  builder     Manage builds
  buildx*     Docker Buildx (Docker Inc., v0.11.2)
  compose*    Docker Compose (Docker Inc., v2.21.0)
  container   Manage containers
  context     Manage contexts
  image       Manage images
  manifest    Manage Docker image manifests and manifest lists
  network     Manage networks
  plugin      Manage plugins
  system      Manage Docker
  trust       Manage trust on Docker images
  volume      Manage volumes

Swarm Commands:
  swarm       Manage Swarm

Commands:
  attach      Attach local standard input, output, and error streams to a running container
  commit      Create a new image from a container's changes
  cp          Copy files/folders between a container and the local filesystem
  create      Create a new container
  diff        Inspect changes to files or directories on a container's filesystem
  events      Get real time events from the server
  export      Export a container's filesystem as a tar archive
  history     Show the history of an image
  import      Import the contents from a tarball to create a filesystem image
  inspect     Return low-level information on Docker objects
  kill        Kill one or more running containers
  load        Load an image from a tar archive or STDIN
  logs        Fetch the logs of a container
  pause       Pause all processes within one or more containers
  port        List port mappings or a specific mapping for the container
  rename      Rename a container
  restart     Restart one or more containers
  rm          Remove one or more containers
  rmi         Remove one or more images
  save        Save one or more images to a tar archive (streamed to STDOUT by default)
  start       Start one or more stopped containers
  stats       Display a live stream of container(s) resource usage statistics
  stop        Stop one or more running containers
  tag         Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE
  top         Display the running processes of a container
  unpause     Unpause all processes within one or more containers
  update      Update configuration of one or more containers
  wait        Block until one or more containers stop, then print their exit codes

Global Options:
      --config string      Location of client config files (default "/root/.docker")
  -c, --context string     Name of the context to use to connect to the daemon (overrides DOCKER_HOST env var and default context set with "docker context use")
  -D, --debug              Enable debug mode
  -H, --host list          Daemon socket to connect to
  -l, --log-level string   Set the logging level ("debug", "info", "warn", "error", "fatal") (default "info")
      --tls                Use TLS; implied by --tlsverify
      --tlscacert string   Trust certs signed only by this CA (default "/root/.docker/ca.pem")
      --tlscert string     Path to TLS certificate file (default "/root/.docker/cert.pem")
      --tlskey string      Path to TLS key file (default "/root/.docker/key.pem")
      --tlsverify          Use TLS and verify the remote
  -v, --version            Print version information and quit

Run 'docker COMMAND --help' for more information on a command.

For more help on how to use Docker, head to https://docs.docker.com/go/guides/

root@ubuntu-g2-small2-bamdad-1:/home/ubuntu#

**Comment by CoderoidAcademy** [comment]
**and docker images**

root@ubuntu-g2-small2-bamdad-1:/home/ubuntu# docker images
REPOSITORY              TAG            IMAGE ID       CREATED         SIZE
appwrite/appwrite       1.4.9          771bcaaf7894   3 weeks ago     1GB
openruntimes/executor   0.4.5          82c16d17eb90   4 weeks ago     273MB
appwrite/assistant      0.2.2          bc0c3e6188da   8 weeks ago     1.3GB
hello-world             latest         9c7a54a9a43c   6 months ago    13.3kB
mariadb                 10.7           895b6c8829c3   7 months ago    396MB
traefik                 2.9            d1e26b5f8193   7 months ago    138MB
redis                   7.0.4-alpine   9192ed4e4955   15 months ago   28.5MB
appwrite/influxdb       1.5.0          0cb9ed3bcc11   23 months ago   217MB
appwrite/telegraf       1.4.0          eb185fa1de51   23 months ago   177MB
root@ubuntu-g2-small2-bamdad-1:/home/ubuntu#

**Comment by CoderoidAcademy** [comment]
**and service --status-all**

root@ubuntu-g2-small2-bamdad-1:/home/ubuntu# service --status-all
 [ + ]  acpid
 [ + ]  apparmor
 [ + ]  apport
 [ - ]  console-setup.sh
 [ + ]  cron
 [ - ]  cryptdisks
 [ - ]  cryptdisks-early
 [ + ]  dbus
 [ + ]  docker
 [ - ]  grub-common
 [ - ]  hwclock.sh
 [ - ]  irqbalance
 [ - ]  iscsid
 [ - ]  keyboard-setup.sh
 [ + ]  kmod
 [ - ]  lvm2
 [ - ]  lvm2-lvmpolld
 [ + ]  multipath-tools
 [ - ]  open-iscsi
 [ - ]  open-vm-tools
 [ + ]  plymouth
 [ + ]  plymouth-log
 [ + ]  procps
 [ + ]  qemu-guest-agent
 [ + ]  resolvconf
 [ - ]  rsync
 [ - ]  screen-cleanup
 [ + ]  ssh
 [ + ]  udev
 [ + ]  ufw
 [ + ]  unattended-upgrades
 [ - ]  uuidd
 [ + ]  watchdog
 [ - ]  wd_keepalive
root@ubuntu-g2-small2-bamdad-1:/home/ubuntu#

**Comment by CoderoidAcademy** [comment]
**the file of this error in appwrite repo**

https://github.com/appwrite/appwrite/blob/5a715ff68cba4171bf3473eabc83b84a4ae5876c/app/http.php

![Capture333](https://github.com/appwrite/appwrite/assets/33560157/c19baf5d-a9bd-49f3-9067-57a9eecf813f)

**Comment by CoderoidAcademy** [comment]
ple

> @kelton-glitch, thanks for raising this issue! 🙏🏼 I don't think the problem is ports, docker, or connecting...It may be related to having an older instance before.
> 
> Would you please share the docker logs from the appwrite container? There should be something related to the 500 error.

**please help us.**

**Comment by kelton-glitch** [comment]
Hello, as an update, you were right @stnguyen90 

I checked again and the WSL installation had some problems that didn't permit Docker to run the appwrite installation properly
I uninstalled the WSL version I had and reinstalled, reconfigured the appwrite port to it's default 8080 and did not get the problem anymore.

**Comment by CoderoidAcademy** [comment]
> Hello, as an update, you were right @stnguyen90
> 
> I checked again and the WSL installation had some problems that didn't permit Docker to run the appwrite installation properly I uninstalled the WSL version I had and reinstalled, reconfigured the appwrite port to its default 8080, and did not get the problem anymore.

Not work this solution. wsl in linux server?!

**Comment by dwaipayan7** [comment]
Container appwrite-redis  Running
 Container openruntimes-executor  Running
 Container appwrite-mariadb  Running
 Container appwrite-task-maintenance  Running
 Container appwrite-assistant  Running
 Container appwrite-task-scheduler-functions  Running
 Container appwrite-worker-messaging  Running
 Container appwrite-worker-certificates  Running
 Container appwrite-worker-mails  Running
 Container appwrite-worker-usage  Running
 Container appwrite-worker-migrations  Running
 Container appwrite-worker-databases  Running
 Container appwrite-worker-deletes  Running
 Container appwrite-task-scheduler-messages  Running
 Container appwrite-worker-usage-dump  Running
 Container appwrite-worker-audits  Running
 Container appwrite  Running
 Container appwrite-realtime  Running
 Container appwrite-worker-webhooks  Running
 Container appwrite-worker-builds  Running
 Container appwrite-worker-functions  Running
 Container appwrite-traefik  Recreate
 Container appwrite-traefik  Recreated
 Container appwrite-traefik  Starting
Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:80 -> 0.0.0.0:0: listen tcp 0.0.0.0:80: bind: An attempt was made to access a socket in a way forbidden by its access permissions.

Getting this error while trying to install the appwrite ...

**Comment by byawitz** [comment]
Hi @dwaipayan7, check your local 80 port. It seems like it's already in use.
